crates/node/src/init.rs

```diff
@@ -164,6 +168,10 @@ impl<'a> NodeInit<'a> {
                     match err {
                         JoinError::MissingNodesConfiguration => true,
                         JoinError::ConcurrentNodeRegistration(_) => false,
+                        JoinError::MetadataStore(ReadWriteError::RemoteError(RemoteError::Unimplemented)) => {
+                            error!("Received an invalid response from the metadata store. This could indicate a misconfigured client or a missing metadata-store role on one of the stores.");
```

Comment:
Should we maybe include the address of the node that we tried to contact so that finding the misconfiguration is easier?

Reply:
I think we are missing the address of the node that failed.

Do we only want to print an error message if the `Code` is unimplemented? Are there other scenarios that point towards a misconfiguration?

Aren't we logging the error also when `NodeInit::init()` fails in `Node`? How would this look like with the additional error message here? Could this be unified into a single place?